### PR TITLE
ENH: Make filter type strings case-insensitive. Closes #1454.

### LINKED ISSUE #1454: Make fbp_op filter_type case insensitive
We currently distinguish between e.g. `'Hann'` and `'hann'`, this is simply irritating and should be fixed.

### PR DESCRIPTION
**Pull request by sbanert**

### CODE REVIEW

**Comment by sbanert** [comment]
I suspect that something is wrong with @pep8speaks.

**Comment by adler-j** [comment]
Basically they changed a standard and now anyone who changes a file gets yelled at. Perhaps we should just make a PR that fixes this globally.

File: odl/tomo/analytic/filtered_back_projection.py [hunk 1]
```diff
@@ -79,16 +80,16 @@ def _fbp_filter(norm_freq, filter_type, frequency_scaling):
     """
     if callable(filter_type):
         filt = filter_type(norm_freq)
-    elif filter_type == 'Ram-Lak':
+    elif filter_type.upper() == 'RAM-LAK':
```
Comment (adler-j) [inline odl/tomo/analytic/filtered_back_projection.py]: prefer `lower`. Uppercase is usually reserved for global constants and could be confusing.

Comment (sbanert) [reply odl/tomo/analytic/filtered_back_projection.py]: Ok. I thought about these strings more like global constants (or an enumeration type). But you're the boss. :smiley:

**Comment by sbanert** [comment]
Regarding @pep8speaks: It complains about both breaks after a binary operator (W504) and breaks before an operator (W503). I think that the latter one is a bug.

File: odl/tomo/analytic/filtered_back_projection.py (hunk 1 above)
Comment (adler-j) [reply odl/tomo/analytic/filtered_back_projection.py]: ODL is a collaborative effort, I try avoiding being a boss the best I can :)

Anyway, I guess the above explanation was not perfect honestly, but I've always seen people use `str.lower()` to normalize capitalization, I tried googling it but didn't see any official guideline on it so perhaps it is just convention.

**Comment by adler-j** [comment]
Feel free to merge

**Comment by kohr-h** [comment]
> Regarding @pep8speaks: It complains about both breaks after a binary operator (W504) and breaks before an operator (W503). I think that the latter one is a bug.

The W503 complains about the "current" way of breaking lines, right? Could you add it to the [`.pep8speaks.yml` ignore section](https://github.com/odlgroup/odl/blob/master/.pep8speaks.yml#L15) then please?

File: odl/tomo/analytic/filtered_back_projection.py [hunk 2]
```diff
@@ -57,6 +57,7 @@ def _fbp_filter(norm_freq, filter_type, frequency_scaling):
                    callable}
         The type of filter to be used.
         If a string is given, use one of the standard filters with that name.
+        Strings are case-insensitive.
```
Comment (kohr-h) [inline odl/tomo/analytic/filtered_back_projection.py]: We never say that explicitly.

**Comment by sbanert** [comment]
Everything done and rebased.

**Pull request merged**